core/types/calls.cc

```diff
@@ -1924,14 +1924,37 @@ class Object_class : public IntrinsicMethod {
     void apply(const GlobalState &gs, const DispatchArgs &args, DispatchResult &res) const override {
         auto mustExist = true;
         ClassOrModuleRef self = unwrapSymbol(gs, args.thisType, mustExist);
+        auto tClassSelfType = Types::tClass(args.selfType);
+        if (self.data(gs)->isModule()) {
+            // This normally can't happen, because `Object` is not an ancestor of any module
+            // instance by default. But Sorbet supports requires ancestor in a really weird way (by
```

Comment:
Does this comment mean this will only happen when [`--enable-experimental-requires-ancestor`](https://sorbet.org/docs/requires-ancestor) is enabled? If so, would it be worth adding an enforce that it's on?

Reply:
i added, with a note that it's not required for the correctness of the change, but simply because we couldn't craft a test case when it would not be true.